File: 30 Days of Code/Day 28 RegEx, Patterns, and Intro to Databases/Email.java
import java.util.*;

public class Email {
	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		int names = input.nextInt();
		input.skip("\n");
		ArrayList<String> arr = new ArrayList<>();

		for (int name = 0; name < names; name++) {
			String email = input.nextLine();
			Pattern gmail = Pattern.compile("@gmail.com");
			Matcher match = gmail.matcher(email);

			if (match.find()) {
				arr.add(email.split(" ")[0]);
			}
		}

		Collections.sort(arr);

		for (String name : arr) {
			System.out.println(name);
		}
	}
}
			
			

File: 30 Days of Code/Day 28 RegEx, Patterns, and Intro to Databases/email.cpp
#include <algorithm>
#include <iostream>
#include <regex>
#include <vector>

using namespace std;

int main()
{
	int names = 0;
	cin >> names;
	cin.get();

	vector<string> gmails;

	for (int i = 0; i < names; i++) {
		string email = "";
		getline(cin, email);
		regex pattern("@gmail.com");

		if (regex_search(email, pattern)) {
			gmails.push_back(email.substr(0, email.find(" ")));
		}
	}

	sort(gmails.begin(), gmails.end());

	for (int i = 0; i < gmails.size(); i++) {
		cout << gmails[i] << endl;
	}
}